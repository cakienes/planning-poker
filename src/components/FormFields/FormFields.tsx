import React from 'react';
import Input from './Input/Input';
import IFormFieldsProps from './interface/IFormFieldsProps';
import Textarea from './Textarea/Textarea';

const FormFields = (props: IFormFieldsProps) => {
    switch (props.type) {
        case 'text':
        case 'number':
            return <Input {...props} />;
        case 'textarea':
            return <Textarea {...props} />;
        default:
            break;
    }
};

export default FormFields;
