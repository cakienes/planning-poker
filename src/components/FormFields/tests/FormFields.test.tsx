import * as React from 'react';
import { createRenderer, ShallowRenderer } from 'react-test-renderer/shallow';
import FormFields from '../FormFields';

describe('FormField', () => {
    let renderer: ShallowRenderer;
    let formFieldsProps: any = null;

    beforeEach(() => {
        renderer = createRenderer();
        formFieldsProps = {
            input: {
                name: null,
                onBlur: null,
                onChange: null,
                onDragStart: null,
                onDrop: null,
                onFocus: null,
                value: null,
            },
            meta: {
                asyncValidating: null,
                autofilled: null,
                dirty: null,
                dispatch: null,
                form: null,
                initial: null,
                invalid: null,
                pristine: null,
                submitFailed: null,
                submitting: null,
                touched: null,
                valid: null,
                visited: null,
            },
            placeholder: null,
            type: null,
        };
    });

    it('rendered correctly FormField while type is text', () => {
        formFieldsProps = {
            ...formFieldsProps,
            type: 'text',
        };
        // @ts-ignore https://github.com/microsoft/TypeScript/issues/21699
        expect(renderer.render(<FormFields {...formFieldsProps} />)).toMatchSnapshot();
    });

    it('rendered correctly FormField while type is number', () => {
        formFieldsProps = {
            ...formFieldsProps,
            type: 'number',
        };
        // @ts-ignore https://github.com/microsoft/TypeScript/issues/21699
        expect(renderer.render(<FormFields {...formFieldsProps} />)).toMatchSnapshot();
    });

    it('rendered correctly FormField while type is textarea', () => {
        formFieldsProps = {
            ...formFieldsProps,
            type: 'textarea',
        };
        // @ts-ignore https://github.com/microsoft/TypeScript/issues/21699
        expect(renderer.render(<FormFields {...formFieldsProps} />)).toMatchSnapshot();
    });

    it('rendered correctly FormField while type is textarea', () => {
        formFieldsProps = {
            ...formFieldsProps,
            type: 'type',
        };
        // @ts-ignore https://github.com/microsoft/TypeScript/issues/21699
        expect(renderer.render(<FormFields {...formFieldsProps} />)).toMatchSnapshot();
    });
});
