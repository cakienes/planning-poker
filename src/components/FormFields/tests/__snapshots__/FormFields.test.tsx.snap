// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField rendered correctly FormField while type is number 1`] = `
<Input
  input={
    Object {
      "name": null,
      "onBlur": null,
      "onChange": null,
      "onDragStart": null,
      "onDrop": null,
      "onFocus": null,
      "value": null,
    }
  }
  meta={
    Object {
      "asyncValidating": null,
      "autofilled": null,
      "dirty": null,
      "dispatch": null,
      "form": null,
      "initial": null,
      "invalid": null,
      "pristine": null,
      "submitFailed": null,
      "submitting": null,
      "touched": null,
      "valid": null,
      "visited": null,
    }
  }
  placeholder={null}
  type="number"
/>
`;

exports[`FormField rendered correctly FormField while type is text 1`] = `
<Input
  input={
    Object {
      "name": null,
      "onBlur": null,
      "onChange": null,
      "onDragStart": null,
      "onDrop": null,
      "onFocus": null,
      "value": null,
    }
  }
  meta={
    Object {
      "asyncValidating": null,
      "autofilled": null,
      "dirty": null,
      "dispatch": null,
      "form": null,
      "initial": null,
      "invalid": null,
      "pristine": null,
      "submitFailed": null,
      "submitting": null,
      "touched": null,
      "valid": null,
      "visited": null,
    }
  }
  placeholder={null}
  type="text"
/>
`;

exports[`FormField rendered correctly FormField while type is textarea 1`] = `
<Textarea
  input={
    Object {
      "name": null,
      "onBlur": null,
      "onChange": null,
      "onDragStart": null,
      "onDrop": null,
      "onFocus": null,
      "value": null,
    }
  }
  meta={
    Object {
      "asyncValidating": null,
      "autofilled": null,
      "dirty": null,
      "dispatch": null,
      "form": null,
      "initial": null,
      "invalid": null,
      "pristine": null,
      "submitFailed": null,
      "submitting": null,
      "touched": null,
      "valid": null,
      "visited": null,
    }
  }
  placeholder={null}
  type="textarea"
/>
`;

exports[`FormField rendered correctly FormField while type is textarea 2`] = `null`;
